src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,461 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.6.12;
+
+interface D3MPoolLike {
+    function deposit(uint256) external;
+    function withdraw(uint256) external;
+    function transfer(address, uint256) external returns (bool);
+    function transferAll(address) external returns (bool);
+    function accrueIfNeeded() external;
+    function assetBalance() external returns (uint256);
+    function maxDeposit() external view returns (uint256);
+    function maxWithdraw() external view returns (uint256);
+    function active() external view returns (bool);
+}
+
+interface D3MPlanLike {
+    function getTargetAssets(uint256) external view returns (uint256);
+    function active() external view returns (bool);
+}
+
+interface VatLike {
+    function debt() external view returns (uint256);
+    function hope(address) external;
+    function ilks(bytes32) external view returns (uint256, uint256, uint256, uint256, uint256);
+    function Line() external view returns (uint256);
+    function urns(bytes32, address) external view returns (uint256, uint256);
+    function gem(bytes32, address) external view returns (uint256);
+    function live() external view returns (uint256);
+    function slip(bytes32, address, int256) external;
+    function move(address, address, uint256) external;
+    function frob(bytes32, address, address, address, int256, int256) external;
+    function grab(bytes32, address, address, address, int256, int256) external;
+    function fork(bytes32, address, address, int256, int256) external;
+    function suck(address, address, uint256) external;
+}
+
+interface EndLike {
+    function debt() external view returns (uint256);
+    function skim(bytes32, address) external;
+}
+
+interface DaiJoinLike {
+    function dai() external view returns (address);
+    function join(address, uint256) external;
+    function exit(address, uint256) external;
+}
+
+interface TokenLike {
+    function approve(address, uint256) external returns (bool);
+}
+
+contract DssDirectDepositHub {
+
+    // --- Auth ---
+    mapping (address => uint256) public wards;
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+    modifier auth {
+        require(wards[msg.sender] == 1, "DssDirectDepositHub/not-authorized");
+        _;
+    }
+
+    enum Mode { NORMAL, MODULE_CULLED, MCD_CAGED }
+
+    VatLike      public immutable vat;
+    DaiJoinLike  public immutable daiJoin;
+    address      public           vow;
+    EndLike      public           end;
+
+    struct Ilk {
```

Comment:
Let's have distinct external getters for each of these values for offchain/ext use.

Reply:
added: d852ed7

@brianmcmichael any better suggestions on naming? doing just `function pool(ilk)` was giving me variable shadow warnings wherever i use `pool` currently so wanted to avoid that

Reply:
I'm not sure we really need the getters here. We did for example for the new `Vat` but that is a much more core contract that is more probably to be queried. Anyway not opposed either, just wondering if deservers the extension of the code.
Btw I'd choose to remove the `ilk` part of the getters and just rename the variables `pool` of the params of the functions. Maybe just to `pool_`

Reply:
@brianmcmichael I think this was your suggestion to add. Does @gbalabasquer 's point change your mind? if not, then I can update the function names per his comment

Reply:
follow: https://github.com/makerdao/dss-vest/blob/master/src/DssVest.sol#L82-L112

Reply:
addressed in https://github.com/makerdao/dss-direct-deposit/pull/65